smslibrary/src/main/java/com/eis/smslibrary/SMSReceivedBroadcastReceiver.java

```diff
@@ -7,14 +7,17 @@
 import android.os.Bundle;
 import android.telephony.SmsMessage;
 import android.util.Log;
+
+import java.util.HashMap;
+import java.util.Map;
+
 import it.lucacrema.preferences.PreferencesManager;
 
 /**
  * Broadcast receiver for received messages, called by Android.
  *
- * @author Luca Crema, Marco Mariotto
+ * @author Luca Crema, Marco Mariotto, Giovanni Velludo
  * @since 29/11/2019
- *
  */
 public class SMSReceivedBroadcastReceiver extends BroadcastReceiver {
```

Comment:
in onReceive Message recostruction based on source phone number is very weak
if i send a multipart and a message you don't notice the difference cause is build to work with uman ends for the communication
it's better to use some kind of sequence id

Reply:
> if i send a multipart and a message you don't notice the difference cause is build to work with uman ends for the communication

What do you mean with "you don't notice the difference"?

> it's better to use some kind of sequence id

That's what multipart messages use, because it's not guaranteed that they'll be delivered in the correct order.

Reply:
if they use a sequence id , length of the sequence and position in the sequece
it's perfect and I misunderstood something

Reply:
[Here](https://en.wikipedia.org/wiki/Concatenated_SMS#Sending_a_concatenated_SMS_using_a_User_Data_Header)'s an explanation of how they work

Reply:
ok
you can mark this as resolved